content/general/push/publish.textile

```diff
@@ -141,12 +141,16 @@ channel.publish({ name: 'example', data: 'data', extras: extras }, function(err)
 
 h2(#direct-publishing). Direct publishing
 
+Ably provides an API that allows native push notifications to be delivered directly to:
+
 * Devices identified by their unique device ID
 * Devices identified by their assigned "@clientId@":/realtime/authentication#identified-clients
-* Devices identified by their native recipient attributes such as their unique @registrationToken@ in the case of GCM, @deviceToken@ in the case of APNS, or @targetUrl@ and @encryptionKey@ in the case of a Web device. This means is particularly useful when migrating to Ably with existing push notification target devices.
+* Devices identified by their native recipient attributes such as their unique @registrationToken@ in the case of GCM, @deviceToken@ in the case of APNS, or @targetUrl@ and @encryptionKey@ in the case of a Web device. This is particularly useful when migrating to Ably with existing push notification target devices.
 
 See the "push admin publish documentation":/general/push/admin#publish for the client library API details, and the "raw push publish REST API documentation":/rest-api#push-publish for information on the underlying direct publishing endpoint used by the client libraries.
 
+Notably, the "push admin":/general/push/admin is only available in our Ruby and JavaScript libraries. If you need to publish to devices with another library, you should use "channel-based publishing":#channel-broadcast.
```

Comment:
Sadly, this is not true.  It is supported in pretty much all of our libraries now, PHP, Python, Node, JS, Ruby, iOS, Android, Java etc.  The only libs that don't support it are Go and .NET, and the latter will add support soonish.